extensions/notebook-renderers/src/textHelper.ts

```diff
@@ -91,22 +93,46 @@ function truncatedArrayOfString(id: string, buffer: string[], linesLimit: number
 
 function scrollableArrayOfString(id: string, buffer: string[], trustHtml: boolean) {
 	const element = document.createElement('div');
-	if (buffer.length > 5000) {
+	if (buffer.length > softScrollableLineLimit) {
 		element.appendChild(generateNestedViewAllElement(id));
 	}
 
-	element.appendChild(handleANSIOutput(buffer.slice(-5000).join('\n'), trustHtml));
+	element.appendChild(handleANSIOutput(buffer.slice(-1 * softScrollableLineLimit).join('\n'), trustHtml));
 
 	return element;
 }
 
-export function createOutputContent(id: string, outputs: string[], linesLimit: number, scrollable: boolean, trustHtml: boolean): HTMLElement {
+export function createOutputContent(id: string, outputText: string, linesLimit: number, scrollable: boolean, trustHtml: boolean): HTMLElement {
 
-	const buffer = outputs.join('\n').split(/\r\n|\r|\n/g);
+	const buffer = outputText.split(/\r\n|\r|\n/g);
 
 	if (scrollable) {
 		return scrollableArrayOfString(id, buffer, trustHtml);
 	} else {
 		return truncatedArrayOfString(id, buffer, linesLimit, trustHtml);
 	}
 }
+
+const outputLengths: Record<string, number> = {};
+
+export function appendScrollableOutput(element: HTMLElement, id: string, appended: string, fullText: string, trustHtml: boolean) {
+	if (!outputLengths[id]) {
+		outputLengths[id] = 0;
+	}
+
+	const buffer = appended.split(/\r\n|\r|\n/g);
+	const appendedLength = buffer.length + outputLengths[id];
+	// Allow the output to grow to the hard limit then replace it with the last softLimit number of lines if it grows too large
```

Comment:
`Allow the displayed output to grow to the hard limit then replace it with the last softLimit number of lines if it grows too large`

Initially on reading the code I was worried you reverted to the old behaviour once the line limit was reached, then I realised your `outputLengths` is actually counting the number of lines displayed in the DOM, and the only expensive update is when this exceeds 8000 at which point you reduce back to displaying 5000.